### PR TITLE
concurrent multipart s3 upload/download, lz4 compression, client tool for json input

### PR DESCRIPTION
**Pull request by umeshdangat**
* client tool can now accept indexing docs from json files, 1 json doc per line
* use lz4 for compression
* multipart s3 upload and parallel s3 download

lz4 vs gzip speed comparisons for a 12G index.

```
umesh@devbox:~$ ls -ltrh /nail/home/umesh/nrtsearch/compressed.*
-rw-r--r-- 1 umesh users 9.4G Feb 21 12:26 /nail/home/umesh/nrtsearch/compressed.lz4
-rw-r--r-- 1 umesh users 7.5G Feb 21 12:40 /nail/home/umesh/nrtsearch/compressed.gzip
umesh@devbox:~$ du -sh /nail/home/umesh/nrtsearch/primary_index/
12G     /nail/home/umesh/nrtsearch/primary_index/
umesh@devbox:~/pg/platypus (client_add_docs_from_json_files) $ ./tar_impl.sh
buildTar with lz4 took 37 seconds
buildTar with gzip took 848 seconds
umesh@devbox:~/pg/platypus (client_add_docs_from_json_files) $ ./tar_extract_impl.sh
extractTar with lz4 took 59 seconds
extractTar with gzip took 105 seconds
```

* parallel s3 upload took around ~3 mins for 12G index
* parallel s3 download takes around ~10seconds per GB of data

### CODE REVIEW

File: src/test/java/org/apache/platypus/server/grpc/GrpcServer.java [hunk 1]
```diff
@@ -13,6 +14,8 @@
 import org.apache.commons.csv.CSVParser;
 import org.apache.platypus.server.luceneserver.GlobalState;
 import org.apache.platypus.server.utils.Archiver;
+import org.apache.platypus.server.utils.ArchiverImpl;
```
Comment (sarthakn7) [inline src/test/java/org/apache/platypus/server/grpc/GrpcServer.java]: Additional imports in this class seem to be added by mistake

File: src/main/java/org/apache/platypus/server/utils/TarImpl.java [hunk 2]
```diff
@@ -126,4 +145,30 @@ private static void addFilestoTarGz(String filePath, String parent, TarArchiveOu
             }
         }
     }
+
+    public static void main(String[] args) throws IOException {
```
Comment (sarthakn7) [inline src/main/java/org/apache/platypus/server/utils/TarImpl.java]: Would it be better to have this in tests (even though it doesn't test with asserts) ?

File: src/test/java/org/apache/platypus/server/grpc/GrpcServer.java (hunk 1 above)
Comment (umeshdangat) [reply src/test/java/org/apache/platypus/server/grpc/GrpcServer.java]: removed

File: src/main/java/org/apache/platypus/server/utils/TarImpl.java (hunk 2 above)
Comment (umeshdangat) [reply src/main/java/org/apache/platypus/server/utils/TarImpl.java]: moved to TarImplTest

**Pull request merged**